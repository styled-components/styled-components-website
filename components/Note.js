import styled from 'styled-components';

import rem from '../utils/rem';
import { blmLightGrey, blmGrey } from '../utils/colors';
import { bodyFont, headerFont } from '../utils/fonts';

import { SubHeader, Title } from './Layout';

export const Note = styled.div`
  font-family: ${bodyFont};
  background: ${blmLightGrey};
  padding: ${rem(7)} ${rem(10)} ${rem(5)} ${rem(14)};
  border-left: ${rem(4)} solid ${blmGrey};
  margin: ${rem(45)} 0;
  border-radius: ${rem(3)};

  > p {
    margin: 0 0 ${rem(5)} 0;
  }

  ${SubHeader} + &, ${Title} + & {
    margin-top: ${rem(35)};
  }
`;

const NoteLabel = styled.strong`
  display: block;
  font-weight: 600;
  font-family: ${headerFont};
  text-transform: uppercase;
  font-size: 90%;
  margin-bottom: ${rem(7)};
`;

const NoteWrapper = ({ label = 'Note', children }) => (
  <Note>
    <NoteLabel>{label}</NoteLabel>
    {children}
  </Note>
);

export default NoteWrapper;
