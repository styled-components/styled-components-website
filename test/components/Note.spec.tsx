import React from 'react';
import renderer from 'react-test-renderer';
import Note from '../../components/Note';

test('Note renders correctly', () => {
  const tree = renderer.create(<Note />).toJSON();

  expect(tree).toMatchSnapshot();
});
