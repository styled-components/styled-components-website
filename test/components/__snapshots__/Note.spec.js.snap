// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  background: #e0e0e0;
  padding: 0.3888888888888889rem 0.5555555555555556rem 0.2777777777777778rem 0.7777777777777778rem;
  border-left: 0.2222222222222222rem solid rgb(33,33,33);
  margin: 2.5rem 0;
  border-radius: 0.16666666666666666rem;
}

.c1 > p {
  margin: 0 0 0.2777777777777778rem 0;
}

.sc-eCssSg + .c0,
.sc-dlfnbm + .c1 {
  margin-top: 1.9444444444444444rem;
}

.c2 {
  display: block;
  font-weight: 600;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  text-transform: uppercase;
  font-size: 90%;
  margin-bottom: 0.3888888888888889rem;
}

<div
  className="c0 c1"
>
  <strong
    className="c2"
  >
    Note
  </strong>
</div>
`;
